lib/image_optim/cmd.rb

```diff
@@ -44,6 +80,42 @@ def check_status!
 
         fail SignalException, status.termsig
       end
+
+      def init_options!(args)
+        pgroup_opt = Gem.win_platform? ? :new_pgroup : :pgroup
+
+        if args.last.is_a?(Hash)
+          args.last[pgroup_opt] = true
+        else
+          args.push(pgroup_opt => true)
+        end
+      end
+
+      def cleanup_process(pid)
+        Thread.new do
+          begin
+            Process.kill('-TERM', pid)
+            Process.detach(pid)
+            now = Time.now
+
+            while Time.now - now < 10
```

Comment:
@toy  10 is sort of a magic number here. Any ideas about how we can handle this better?